FirebaseAuth/Sources/Backend/FIRAuthBackend.m

```diff
@@ -846,9 +856,17 @@ - (void)verifyPassword:(FIRVerifyPasswordRequest *)request
 #if TARGET_OS_IOS
                    NSMutableArray<FIRMultiFactorInfo *> *multiFactorInfo = [NSMutableArray array];
                    for (FIRAuthProtoMFAEnrollment *MFAEnrollment in response.MFAInfo) {
-                     FIRPhoneMultiFactorInfo *info =
-                         [[FIRPhoneMultiFactorInfo alloc] initWithProto:MFAEnrollment];
-                     [multiFactorInfo addObject:info];
+                     // check which MFA factors are enabled.
+                     if (MFAEnrollment.phoneInfo != nil) {
+                       FIRPhoneMultiFactorInfo *info =
+                           [[FIRPhoneMultiFactorInfo alloc] initWithProto:MFAEnrollment];
+                       [multiFactorInfo addObject:info];
+                     }
```

Comment:
Should there be an else here? Can a single info be both a phone and a totp?

Reply:
Hmm, it looks like the initializer in [FirebaseAuth/Sources/Backend/RPC/Proto/FIRAuthProtoMFAEnrollment.m](https://github.com/firebase/firebase-ios-sdk/pull/11261/files#diff-046c2c1b9ee232f9ea8fe9d9c614cc4013fd7087d84f4ae641098a9a501e1f69) will make any particular FIRAuthProtoMFAEnrollment either a phoneInfo or a TOTP info.

Reply:
That is correct.  Because of they are implicitly assigned with either phoneInfo or TOTP info, it made sure that only one `if` block will be executed.

However, I do agree adding an else block is less error prone.  Updated the code.